File: 프로그래머스/1/42862. 체육복/체육복.py
def solution(n, lost, reserve):
    answer = 0

    arr = [1] * (n)

    for lost_s in lost:
        arr[lost_s - 1] -= 1
        
    for reserve_s in reserve:
        arr[reserve_s - 1] += 1
                
    for i in range(n):
        if arr[i] == 2:
            if i-1 >= 0 and i-1 < n and arr[i-1] == 0:
                arr[i-1] = 1
                arr[i] = 1
            
            if arr[i] == 2:
                if i+1 >= 0 and i+1 < n and arr[i+1] == 0:
                    arr[i+1] = 1
                    arr[i] = 1
                
    for elem in arr:
        if elem > 0:
            answer += 1

    return answer

File: 프로그래머스/1/42862. 체육복/체육복.cpp
#include <string>
#include <vector>
#include <iostream>

using namespace std;

int solution(int n, vector<int> lost, vector<int> reserve) {
    int answer = 0;
    
    // Step 1: Initialize student items with 1
    vector<int> student(n + 1, 1);
    
    // Step 2: Mark lost items
    for (auto elem : lost) {
        student[elem]--;
    }
    
    // Step 3: Mark reserve items
    for (auto elem : reserve) {
        student[elem]++;
    }
    
    // Step 4: Borrow items if necessary
    for (int i = 1; i <= n; i++) {
        if (student[i] == 0) {
            if (i - 1 > 0 && student[i - 1] == 2) {
                student[i - 1]--;
                student[i]++;
            } else if (i + 1 <= n && student[i + 1] == 2) {
                student[i + 1]--;
                student[i]++;
            }
        }
    }
    
    // Step 5: Count students with at least 1 item
    for (int i = 1; i <= n; i++) {
        if (student[i] > 0) answer++;
    }
    
    return answer;
}
